### PR TITLE
[Inductor] Add triton.autotune support for user defined triton kernels with constant/simple grids

### PR DESCRIPTION
**Pull request by oulgen**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112292
* #112290
* __->__ #112228

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Comment by oulgen** [comment]
This PR only handles simple grids, on the next PR I will deal with complex grids

File: torch/_inductor/triton_heuristics.py [hunk 1]
```diff
@@ -362,16 +363,17 @@ def kernel_call():
                     {**dict(zip(self.arg_names, args)), **launcher.config.kwargs}
```
Comment (oulgen) [inline torch/_inductor/triton_heuristics.py:363]: Hmm, I'm not really sure what this pre_hook stuff is. I assume I need to put `**kwargs` here too, how to trigger/test this?

**Review by jansel — APPROVED** [review]

**Comment by huydhn** [comment]
@pytorchbot revert -m 'Sorry for reverting your change but it is failing ROCm test in trunk https://hud.pytorch.org/pytorch/pytorch/commit/dbb31a2984fa616b4bb6fac7abb2a06ec0533eb1' -c nosignal

You can add `ciflow/rocm` to your PR to trigger the failure.

**Comment by oulgen** [comment]
@huydhn No worries, I just disabled this test under ROCM. If rocm mode is valuable, could we run these tests by default so that we don't have to revert diffs?

**Comment by huydhn** [comment]
> @huydhn No worries, I just disabled this test under ROCM. If rocm mode is valuable, could we run these tests by default so that we don't have to revert diffs?

We have a smaller number of ROCM runners than Linux CUDA.  So this is a trade-off between CI coverage and queuing time.  There is a work in progress https://github.com/pytorch/pytorch/pull/110511 to run only a subset of tests on ROCm for every PR.  Adding dynamo tests in the set could help avoid similar cases.

**Comment by jansel** [comment]
Let's just disable the test on rocm.

**Comment by oulgen** [comment]
@jansel already done and pushed with the disable

**Pull request closed without merging**